march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py

```diff
@@ -84,64 +94,59 @@ def __init__(self, context):
         self.time_slider = self._widget.RvizFrame.findChild(QSlider, "TimeSlider")
         self.scale_setpoints_check_box = self._widget.GaitPropertiesFrame.findChild(QCheckBox, "ScaleSetpoints")
 
-        # Connect Gait settings buttons
-        self.set_gait_directory_button(self.gait_directory)
-        self.change_gait_directory_button.clicked.connect(
-            lambda: [
-                self.set_gait_directory_button(self.get_gait_directory(True))
-            ]
-        )
+        self.connect_buttons()
 
-        self.import_gait_button.clicked.connect(
-            lambda: [
-                self.load_gait()
-            ]
-        )
+    # Called by build_ui
+    def create_rviz_frame(self):
+        frame = rviz.VisualizationFrame()
+        frame.initialize()
+        reader = rviz.YamlConfigReader()
+        config = rviz.Config()
+        reader.readFile(config,
+                        os.path.join(rospkg.RosPack().get_path('march_rqt_gait_generator'), 'resource', 'cfg.rviz'))
+        frame.load(config)
 
-        self.export_gait_button.clicked.connect(self.export)
+        # Hide irrelevant Rviz details
+        frame.setMenuBar(None)
+        frame.setStatusBar(None)
+        frame.setHideButtonVisibility(False)
+        return frame
 
-        self.publish_gait_button.clicked.connect(
-            lambda: self.publish_gait()
-        )
+    # Called by build_ui
+    def connect_buttons(self):
+        self.change_gait_directory_button.clicked.connect(self.change_gait_directory)
+        self.import_gait_button.clicked.connect(self.import_gait)
+        self.export_gait_button.clicked.connect(self.export_gait)
+
+        self.publish_gait_button.clicked.connect(self.publish_gait)
 
         self.start_button.clicked.connect(self.start_time_slider_thread)
         self.stop_button.clicked.connect(self.stop_time_slider_thread)
         self.undo_button.clicked.connect(self.undo)
         self.redo_button.clicked.connect(self.redo)
 
-        self.playback_speed_line_edit.setValidator(QtGui.QIntValidator(0, 500, self))
```

Comment:
This was necessary because playback_speed_line_edit was a line edit, meaning that the input can be any string. This statement limits the input to doubles between 0 and 500. Instead, I changed the playback_speed_line_edit to a spinbox, which only accepts integers in the first place. The maximum of 500 now set in the gait_generator.ui file.